common/src/main/scala/org/apache/comet/vector/NativeUtil.scala

```diff
@@ -107,16 +109,16 @@ class NativeUtil {
             null
           }
 
-          // The array and schema structures are allocated by native side.
-          // Don't need to deallocate them here.
-          val arrowSchema = ArrowSchema.wrap(schemaAddrs(index))
-          val arrowArray = ArrowArray.wrap(arrayAddrs(index))
```

Comment:
This is to follow Arrow spec and is the correct way to use ArrowSchema and ArrowArray. What you changed is back to previous incorrect approach which causes some memory issue before.

Reply:
updated